code/modules/power/groundmap_geothermal.dm

```diff
@@ -114,78 +107,65 @@ GLOBAL_VAR_INIT(generators_on_ground, 0)
+		if(!SSmachines.generators_on_ground) //Prevent division by 0
 			return PROCESS_KILL
 		if((length(GLOB.humans_by_zlevel["2"]) > 0.2 * length(GLOB.alive_human_list_faction[FACTION_TERRAGOV])))
 			//You get points proportional to the % of generators corrupted (for example, if 66% of generators are corrupted the hive gets 0.66 points per second)
-			var/points_generated = GENERATOR_PSYCH_POINT_OUTPUT / GLOB.generators_on_ground
+			var/points_generated = GENERATOR_PSYCH_POINT_OUTPUT / SSmachines.generators_on_ground
 			SSpoints.add_strategic_psy_points(corrupted, points_generated)
 			SSpoints.add_tactical_psy_points(corrupted, points_generated*0.25)
 		return
 	if(!is_on || buildstate || !anchored || !powernet) //Default logic checking
 		return PROCESS_KILL
 
-	if(!check_failure()) //Wait! Check to see if it breaks during processing
-		if(power_gen_percent < 100)
-			power_gen_percent++
-			update_icon()
-			switch(power_gen_percent)
-				if(10)
-					visible_message("[icon2html(src, viewers(src))] [span_notice("<b>[src]</b> begins to whirr as it powers up.")]")
-				if(50)
-					visible_message("[icon2html(src, viewers(src))] [span_notice("<b>[src]</b> begins to hum loudly as it reaches half capacity.")]")
-				if(100)
-					visible_message("[icon2html(src, viewers(src))] [span_notice("<b>[src]</b> rumbles loudly as the combustion and thermal chambers reach full strength.")]")
-		add_avail(power_generation_max * (power_gen_percent / 100) ) //Nope, all good, just add the power
-
-/obj/machinery/power/geothermal/proc/check_failure()
-	cur_tick++
-	if(cur_tick < fail_check_ticks) //Nope, not time for it yet
-		return FALSE
-	else if(cur_tick > fail_check_ticks) //Went past with no fail, reset the timer
-		cur_tick = 0
-		return FALSE
-	if(rand(1,100) < fail_rate) //Oh snap, we failed! Shut it down!
-		if(rand(0,3) == 0)
-			visible_message("[icon2html(src, viewers(src))] <span class='notice'><b>[src]</b> beeps wildly and a fuse blows! Use wirecutters, then a wrench to repair it.")
-			buildstate = GEOTHERMAL_MEDIUM_DAMAGE
-			icon_state = "wire"
-		else
-			visible_message("[icon2html(src, viewers(src))] <span class='notice'><b>[src]</b> beeps wildly and sprays random pieces everywhere! Use a wrench to repair it.")
-			buildstate = GEOTHERMAL_LIGHT_DAMAGE
-			icon_state = "wrench"
-
-		//Resets the fail_rate incase the xenos have been fucking with it.
-		fail_rate = initial(fail_rate)
-
-		is_on = FALSE
-		power_gen_percent = 0
+	if(power_gen_percent < 100)
+		power_gen_percent++
 		update_icon()
-		cur_tick = 0
-		stop_processing()
-		return TRUE
-	return FALSE //Nope, all fine
+		switch(power_gen_percent)
+			if(10)
+				balloon_alert_to_viewers("[src] begins to whirr as it powers up.")
+			if(50)
+				balloon_alert_to_viewers("[src] begins to hum loudly as it reaches half capacity.")
+			if(100)
+				balloon_alert_to_viewers("[src] rumbles loudly as the combustion and thermal chambers reach full strength.")
```

Comment:
dont use name inserts in baloon alerts: players see its above the generator they can see whats doing it